integrationTests/multiShard/smartContract/executingSCCalls_test.go

```diff
@@ -0,0 +1,452 @@
+package smartContract
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/vm"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/stretchr/testify/assert"
+)
+
+var gasPrice = 1
+var gasLimit = 1000
+var initialValueForInternalVariable = uint64(45)
+
+func createScCallsNodes() (p2p.Messenger, []*testNode) {
+	advertiser := createMessengerWithKadDht(context.Background(), "")
+	advertiser.Bootstrap()
+
+	nodes := createNodes(
+		2,
+		1,
+		getConnectableAddress(advertiser),
+	)
+	displayAndStartNodes(nodes)
+
+	return advertiser, nodes
+}
+
+func deploySmartContract(t *testing.T, nodeToProcess *testNode, roundNumber uint32, senderAddressBytes []byte, senderNonce uint64) {
+	scCode := "aaaa"
+
+	contractTx := createTx(
+		t,
+		senderAddressBytes,
+		vm.CreateEmptyAddress().Bytes(),
+		senderNonce,
+		big.NewInt(0),
+		scCode,
+		initialValueForInternalVariable,
+	)
+
+	err := nodeToProcess.txProcessor.ProcessTransaction(contractTx, roundNumber)
+	assert.Nil(t, err)
+	_, err = nodeToProcess.accntState.Commit()
+	assert.Nil(t, err)
+}
+
+func createTx(
+	t *testing.T,
+	senderAddressBytes []byte,
+	receiverAddressBytes []byte,
+	senderNonce uint64,
+	value *big.Int,
+	scCodeOrFunc string,
+	scValue uint64,
+) *transaction.Transaction {
+
+	txData := fmt.Sprintf("%s@%X", scCodeOrFunc, scValue)
+	tx := &transaction.Transaction{
+		Nonce:    senderNonce,
+		Value:    value,
+		RcvAddr:  receiverAddressBytes,
+		SndAddr:  senderAddressBytes,
+		GasPrice: uint64(gasPrice),
+		GasLimit: uint64(gasLimit),
+		Data:     txData,
+	}
+	assert.NotNil(t, tx)
```

Comment:
Do you need this assert here?

Reply:
No, it remained from a different implementation, will remove :)

Reply:
ok